packages/hooks/src/useInputDecimalFormatter.ts

```diff
@@ -21,25 +21,19 @@ const useInputDecimalFormatter = (initial?: number, options?: TOptions) => {
                 const text = with_sign ? new_value : new_value.replaceAll(/[+-]/g, '');
                 const inputs = text.split('.');
 
-                // The field contains more than one dot, So we return the old value as only one dot
-                // is allowed.
-                if (inputs.length > 2) return old_value;
-
                 const left = inputs[0];
                 const right = inputs.length > 1 ? inputs[1] : null;
                 const has_right = right !== null && right !== '';
 
-                // The field value is positive or negative sign, So we return the new value without
+                // The field value is positive or negative sign or 0, So we return the new value without
                 // any calculations.
-                if ((left === '-' || left === '+') && !has_right) return new_value;
-
-                // The field value is 0, So we return the new value without any calculations.
-                if (left === '0' && !has_right) return new_value;
+                if (['-', '+', '0'].includes(left) && !has_right) return new_value;
 
                 const is_number = !isNaN(Number(new_value));
-
-                // The input value is not a valid number, So we return the old value.
-                if (!is_number) return old_value;
+                // The field contains more than one dot, So we return the old value as only one dot
+                // is allowed.
+                // Or if the input value is not a valid number, So we return the old value.
```

Comment:
In my opinion, the previous code following the return early pattern was more readable and maintainable, Making the code shorter by combining the if conditions with `or` doesn't reduce the complexity, It makes it harder to maintain it.

Reply:
Yea I actually don't find this hook to really need any refactors, most of it is already readable, its just that sonarcloud was still complaining about the cognitive complexity of it (even though its already understandable the way I see it), its just that when the conditions are broken apart into different statements, sonarcloud adds a +2 for every statement 🤷

Reply:
Besides the `includes` refactorng in line 28, I think I'll revert this refactor

Reply:
@adrienne-deriv I didn't mean to ignore the sonarcloud here, I meant silencing the error by patching the code is not a good approach. One thing we can do is to break this function into smaller utility functions with fewer and more specific if conditions.